connectors/byoei.py

```diff
@@ -100,7 +108,7 @@ def _bulk_op(self, doc, operation=OP_INDEX):
             return [{operation: {"_index": index, "_id": doc_id}}, doc["doc"]]
         if operation == OP_UPSERT:
             return [
-                {"update": {"_index": index, "_id": doc_id}},
+                {operation: {"_index": index, "_id": doc_id}},
```

Comment:
I think we should encapsulate an operation in a class `Operation`. Would make it more readable and would be more consistent with other parts of the codebase, where we've started to encapsulate plain dicts into classes extending `UserDict`.

Reply:
Agreed, but not in the scope of this PR IMO